docs/modules/ROOT/pages/how-to-guides/proc_prefetching-dependencies-to-support-hermetic-build.adoc

```diff
@@ -0,0 +1,193 @@
+
+[#supported-languages]
+.Supported languages
+
+[cols="1,1"]
+|===
+|Language
+|Package Manager
+
+|Go
+|`gomod`
+
+|Python
+|`pip`
+|===
+
+== Enabling prefetch builds for `gomod`
+
+.Prerequisites
+* You have an upgraded build pipeline.
+* You have a `go.mod` file in your repository that lists all the the dependencies.
+
+.Procedures
+For each component that you want to build hermetically:
+
+. Navigate to your component's source code.
+. Open the *.tekton* directory and locate the *.yaml* files related to the *pull-request* and *push* processes.
+. *Set up dependency prefetching with Cachi2:* In both the `.yaml` files, add the following prefetch-input pipeline parameter.
+
++
+[source,yaml]
+----
+param:
+    -  name: prefetch-input
+       value: gomod <1>
+----
+<1> If the package manager lock file is located in the root directory of your project, enter `gomod`. Otherwise, if the package manager lock file is in a subdirectory, provide the path to the lock file in JSON format. For example,
+{"type": "{<package-manager>}", "path": "path-to-the-directory-which-contains-the-lockfile}"}
+
+include::../partials/con_hermetic_verification.adoc[]
+
+== Enabling prefetch builds for `pip`
+Cachi2 supports pip by parsing of `pip` requirements files, including but not limited to *requirements.txt* files placed in the root of your repository. By generically parsing `pip` requirements files, it downloads the specified dependencies.
+
+IMPORTANT: The requirements file can have a different name, as you can use multiple files to provide the dependencies. These requirements files function as lockfiles, encompassing all the transitive dependencies. It's crucial to actively pin each transitive dependency listed in the requirements file to a specific version.
+
+
+s, which include all the transitive dependencies. Each transitive dependency in the *requirements.txt* file must be pinned to a specific version.
+
+.Prerequisites
+* You have an upgraded build pipeline.
+* You have a <filename> file in your repository that lists all the the dependencies.
+
+.Procedures
+For each component that you want to build hermetically:
+
+. Navigate to your component’s source code.
+. Open the *.tekton* directory and locate the *.yaml* files related to the *pull-request* and *push* processes.
+. *Set up dependency prefetching with Cachi2:* In both the `.yaml` files, add the following prefetch-input pipeline parameter.
+
++
+[source,yaml]
+----
+param:
+    -  name: prefetch-input
+       value: pip <1>
+----
+<1> If the package manager lock file is located in the root directory of your project, enter `pip`. Otherwise, if the package manager lock file is in a subdirectory, provide the path to the lock file in the JSON format. For example,
+{"type": "{<package-manager>}", "path": "path-to-the-directory-which-contains-the-lockfile}"}
+
+. Create a file (for example, pyproject.toml) with metadata (name and version) and place it in the root of your repository.
```

Comment:
I think we're allowed to be directive in our steps to users. 

4. Create a prproject.toml file with metadata, including name and version. 
5. Place this file in the root of your repository: 
[sample metadata you provide]